pkg/reconcilermanager/controllers/reconciler_base.go

```diff
@@ -105,6 +105,10 @@ type reconcilerBase struct {
 
 	// syncKind is the kind of the sync object: RootSync or RepoSync.
 	syncKind string
+
+	// controllerName is used by tests to de-dupe controllers
+	// controller-runtime requires unique names for each controller instantiation
+	controllerName string
```

Comment:
I assume the requirement is new?

But I'm pretty sure the name method has been around. So can you pull this out to a different PR, just to reduce the size of this one?

Reply:
Yes the requirement is new, pulled the change out to a separate PR: https://github.com/GoogleContainerTools/kpt-config-sync/pull/1548